core/services/offchainreporting2/data_source.go

```diff
@@ -29,6 +29,13 @@ type dataSource struct {
 
 var _ median.DataSource = (*dataSource)(nil)
 
+// New creates a copy of dataSource with a new job spec
```

Comment:
spoke offline - lets pull out a separate one which is just in memory for the fee source